cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -360,309 +374,320 @@ TEST_F(TestConvertParquetSchema, ParquetFlatDecimals) {
-    parquet_fields.push_back(
-        GroupNode::Make("my_list", Repetition::OPTIONAL, {array}, ConvertedType::LIST));
-    auto arrow_str = ::arrow::field("str", UTF8, false);
-    std::vector<std::shared_ptr<Field>> fields({arrow_str});
-    auto arrow_struct = ::arrow::struct_(fields);
-    auto arrow_element = ::arrow::field("array", arrow_struct, false);
-    auto arrow_list = ::arrow::list(arrow_element);
-    arrow_fields.push_back(::arrow::field("my_list", arrow_list, true));
-  }
+    // // List<Tuple<String, Integer>> (nullable list, non-null elements)
+    // optional group my_list (LIST) {
+    //   repeated group element {
+    //     required binary str (UTF8);
+    //     required int32 num;
+    //   };
+    // }
+    {
+      auto str_element = PrimitiveNode::Make(
+          "str", Repetition::REQUIRED, ParquetType::BYTE_ARRAY, ConvertedType::UTF8);
+      auto num_element =
+          PrimitiveNode::Make("num", Repetition::REQUIRED, ParquetType::INT32);
+      auto element =
+          GroupNode::Make("element", Repetition::REPEATED, {str_element, num_element});
+      parquet_fields.push_back(GroupNode::Make("my_list", Repetition::OPTIONAL, {element},
+                                               ConvertedType::LIST));
+      auto arrow_str = ::arrow::field("str", UTF8, false);
+      auto arrow_num = ::arrow::field("num", INT32, false);
+      std::vector<std::shared_ptr<Field>> fields({arrow_str, arrow_num});
+      auto arrow_struct = ::arrow::struct_(fields);
+      auto arrow_element = ::arrow::field("element", arrow_struct, false);
+      auto arrow_list = list_case.type_factory(arrow_element);
+      arrow_fields.push_back(::arrow::field("my_list", arrow_list, true));
+    }
 
-  // // List<OneTuple<String>> (nullable list, non-null elements)
-  // optional group my_list (LIST) {
-  //   repeated group my_list_tuple {
-  //     required binary str (UTF8);
-  //   };
-  // }
-  // Special case: group named ends in _tuple
-  {
-    auto element = PrimitiveNode::Make("str", Repetition::REQUIRED,
-                                       ParquetType::BYTE_ARRAY, ConvertedType::UTF8);
-    auto array = GroupNode::Make("my_list_tuple", Repetition::REPEATED, {element});
-    parquet_fields.push_back(
-        GroupNode::Make("my_list", Repetition::OPTIONAL, {array}, ConvertedType::LIST));
-    auto arrow_str = ::arrow::field("str", UTF8, false);
-    std::vector<std::shared_ptr<Field>> fields({arrow_str});
-    auto arrow_struct = ::arrow::struct_(fields);
-    auto arrow_element = ::arrow::field("my_list_tuple", arrow_struct, false);
-    auto arrow_list = ::arrow::list(arrow_element);
-    arrow_fields.push_back(::arrow::field("my_list", arrow_list, true));
-  }
+    // // List<OneTuple<String>> (nullable list, non-null elements)
+    // optional group my_list (LIST) {
+    //   repeated group array {
+    //     required binary str (UTF8);
+    //   };
+    // }
+    // Special case: group is named array
```

Comment:
Emmm this is special case in parquet list spec, why do we rename this?

Reply:
This code is just reindented, but the diff view messes things up unfortunately.

Reply:
ohhh, my bad